libraries/AP_RangeFinder/AP_RangeFinder_LightWareI2C.cpp

```diff
@@ -84,7 +87,12 @@ bool AP_RangeFinder_LightWareI2C::get_reading(uint16_t &reading_cm)
     bool ret = _dev->read((uint8_t *) &val, sizeof(val));
     if (ret) {
         // combine results into distance
-        reading_cm = be16toh(val);
+        uint16_t distance = be16toh(val);
+        if (distance <= _instanceState.max_distance_cm) {
+            reading_cm = distance;
+        } else {
+            ret = false;
```

Comment:
This code is bad for two reasons:

- the check for the defined max distance is already done when `update_status` is called and it sets the correct status of `RangeFinder_OutOfRangeHigh`
- returning false means there is no connection, status will be set to `RangeFinder_NoData`, which is wrong.

Reply:
@OXINARF san..
I will do the same processing as VL53L0X.
I change the update time when I get a normal value.
I do not change the distance and update time when I get the value exceeding the maximum value. I will return true to the return value.
If I did not change past the update time by 200 ms, I set the return value to false.